service/service_test.go

```diff
@@ -515,3 +515,15 @@ func TestAuthenticateUser(t *testing.T) {
 		}
 	}
 }
+
+func TestNewSecret(t *testing.T) {
+	counter := 4
+	for counter != 0 {
+		s := Service{}
+		response, err := s.NewSecret(context.TODO(), nil)
+		assert.Nil(t, err)
+		assert.Equal(t, codes.OK.String(), response.Message)
+
+		counter--
```

Comment:
There is no need to perform a check in the db here using a DAO because this function is meant to be called by a client.